### PR TITLE
fixes #329 - add root path spa support to router

### LINKED ISSUE #329: Make sure Router and API work with root path SPA
If deployed SPA to root path / we need to make sure the router and API works.

### PR DESCRIPTION
**Pull request by Jared-Sprague**
Adding root path spa support to router.  

Fixes #329

### CODE REVIEW

File: packages/router/.gitignore [hunk 1]
```diff
@@ -61,4 +61,7 @@ typings/
 .next
 
 # WebStorm
-.idea
\ No newline at end of file
+.idea
```
Comment (kunyan) [inline packages/router/.gitignore:64]: I think the `.gitignore` in sub package should be removed. it won't work, right ?

File: packages/router/index.js [hunk 2]
```diff
@@ -6,67 +6,99 @@ const { log } = require("@spaship/common/lib/logging/pino");
...
 
-const customRouter = function(req) {
+const customRouter = function (req) {
   let url = req.url.replace(/\/+/g, "/"); // Avoid duplicate slash issue
-  let routeUrl;
-  let match;
-  let spaPath;
-  let matchedFlatDir;
+  let routeUrl; // This is the final path that the router will route the request to
+  let regExMatch; // Regular expression match object for finding SPA paths in the url
```
Comment (kunyan) [inline packages/router/index.js:32]: If a variable should not change, you should consider to replace `let` to `const`

File: packages/router/.gitignore (hunk 1 above)
Comment (sayak-sarkar) [reply packages/router/.gitignore:64]: @kunyan It'd work. It will simply override the root `.gitignore`.

**Comment by sayak-sarkar** [comment]
The PR looks good to me. Once `index.test.js` is merged this should be good to go.

File: packages/router/.gitignore (hunk 1 above)
Comment (Jared-Sprague) [reply packages/router/.gitignore:64]: I'll remove it from the sub package

File: packages/router/index.js (hunk 2 above)
Comment (Jared-Sprague) [reply packages/router/index.js:32]: Agreed, all of these variables change though so they have to be `let`

File: packages/router/.gitignore (hunk 1 above)
Comment (Jared-Sprague) [reply packages/router/.gitignore]: fixed

**Comment by Jared-Sprague** [comment]
Ok fixed the tests, and added an extra test for ROOTSPA.  Merging now.

**Pull request merged**